impl/receiving_requests.go

```diff
@@ -251,8 +265,8 @@ func (m *manager) recordAcceptedValidationEvents(chst datatransfer.ChannelState,
 	chid := chst.ChannelID()
 
 	// record the voucher result if present
-	if result.VoucherResult != nil {
-		err := m.channels.NewVoucherResult(chid, result.VoucherResult)
+	if result.VoucherResult != nil && result.VoucherResult.Voucher != nil {
```

Comment:
Why would someone pass a VoucherResult that wasn't nil but had a nil voucher? I guess that's a possibility and perhaps best to be defense against code supplied outside the library.